oic.wk.rd.raml

```diff
@@ -0,0 +1,166 @@
+-  rdPublish: !include schemas/oic.rd.publish-schema.json
+	
+/oic/rd:
+  displayName: Resource directory resource
+  description: |
+    Resource to be exposed by any Device that can act as a Resource Directory.
+    1) Provides selector criteria (e.g. integer) with GET request 
+    2) Publish or Update a Link in /oic/res with POST request 
+    3) Delete a Link in /oic/res with DELETE request 
+
+  get:
+    description: |
+      Get the attributes of the Resource Directory for selection purposes.
+    queryParameters:
+      if:
+        description: "The Interface to be used."
+        enum: ["oic.if.baseline"]
+    responses:
+      200:
+        description: |
+          Respond with the selector criteria - either the set of attributes or the bias factor
+        body:
+          application/json:
+            schema: rdSelection
+            example: |
+              {
+                "rt": ["oic.wk.rd"],
+		"if": ["oic.if.baseline"], 
+                "sel": 50
+              }
+
+  post:
+    description: |
+      Publish the resource information for the first time or Update the existing one in /oic/res. 
+      Appropriates parts of the information, i.e. Links of the published Resources will be discovered through /oic/res. 
+      1) When a Device first publishes a Link, the request payload to RD may include the Links without "ins" Parameter.  	 
+      2) Upon granting the request, the RD assigns a unique instance value identifying the Link among all the Links it advertises  
+         and sends back the instance value in "ins" Parameter in the Link to the publishing Device.  
+      3) When later the publishing Device updates the existing Link, i.e. changing its Endpoint information,   
+         the request payload to RD needs to include the instance value in "ins" Parameter to identify the Link to update.  
+    queryParameters:
+      rt:
+        description: "used in POST request to ask the RD to add the Links in payload to /oic/res."
+        enum: ["oic.wk.rdpub"]		
+        example: |
+          POST /oic/rd?rt=oic.wk.rdpub        
+      #"rt" query is used in POST request but such usage is unprecedented & out-of-sync with existing "rt" query practice, 
+      # so I recommend to remove this if IoTivity allows. Take notice here the absence of "rt" would produce the same result.  			
+    body:
+      application/json:
+        schema: rdPublish
+        example: |
+              {
+                "di": "e61c3e6b-9c54-4b81-8ce5-f9039c1d04d9", 
+                "links": [
+                  {
+                    "anchor": "ocf://e61c3e6b-9c54-4b81-8ce5-f9039c1d04d9",                   
+                    "href":   "/myLightSwitch",
+                    "rt":     ["oic.r.switch.binary"],
+                    "if":     ["oic.if.a", "oic.if.baseline"],
+                    "p":      {"bm": 3}, 
+                    "eps": [
+                        {"ep": "coaps://[fe80::b1d6]:1111", "pri": 2},
```

Comment:
I'd recommend that RD examples avoid link local IPv6 addresses.  See https://gerrit.iotivity.org/gerrit/#/c/19131/